From 2d3b928eac96fa601080afba9282cac711edbcef Mon Sep 17 00:00:00 2001
From: Lu Feifei <lufeifei@wxiat.com>
Date: Wed, 15 Jun 2022 14:59:51 +0800
Subject: [PATCH] sw64: add a misc device to chip_vt.dts for memory-hotplug
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 2d3b928eac96fa601080afba9282cac711edbcef
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56WV8

--------------------------------

This is an emulated device used to communicate with the memory-hotplug
device on the qemu side, so the device is added to device tree for
guest os.

Signed-off-by: Lu Feifei <lufeifei@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/boot/dts/chip_vt.dts | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/sw_64/boot/dts/chip_vt.dts b/arch/sw_64/boot/dts/chip_vt.dts
index f0bcf1db1d08..abad29dee97e 100644
--- a/arch/sw_64/boot/dts/chip_vt.dts
+++ b/arch/sw_64/boot/dts/chip_vt.dts
@@ -34,5 +34,17 @@ uart: serial0@8801 {
 			clock-frequency = <24000000>;
 			status = "okay";
 		};
+		misc: misc0@8036 {
+			#address-cells = <2>;
+			#size-cells = <2>;
+			compatible = "sw6,sunway-ged";
+			reg = <0x8036 0x0 0x0 0x20>;
+			interrupt-parent=<&intc>;
+			interrupts = <13>;
+			reg-shift = <0>;
+			reg-io-width = <8>;
+			clock-frequency = <24000000>;
+			status = "okay";
+		};
 	};
 };
-- 
2.33.0

